src/hotspot/share/opto/escape.cpp

```diff
@@ -900,6 +915,7 @@ void ConnectionGraph::add_call_node(CallNode* call) {
         if (length < 0 || length > EliminateAllocationArraySizeLimit) {
           // Not scalar replaceable if the length is not constant or too big.
           scalar_replaceable = false;
+          NOT_PRODUCT(nsr_reason = "has a non-constant or too big length");
```

Comment:
Can you separate these two cases: `non-constant` and `too big`? You can split the above check.

Reply:
Done.